src/sentry/static/sentry/app/components/events/interfaces/threads/threadsSelectorOption.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import styled from '@emotion/styled';
+
+import space from 'app/styles/space';
+import TextOverflow from 'app/components/textOverflow';
+import InlineSvg from 'app/components/inlineSvg';
+import {EntryTypeData} from 'app/types';
+
+type Props = {
+  id: string;
+  name?: string;
+  crashed?: boolean;
+  details: ThreadInfo;
+  crashedInfo?: EntryTypeData;
+};
+
+type ThreadInfo = {
+  label: string;
+  filename?: string;
+};
+
+// TODO (i18n): added translations here
+const ThreadsSelectorOption: React.FC<Props> = ({
+  id,
+  name,
+  details,
+  crashed,
+  crashedInfo,
+}) => (
+  <Wrapper>
+    <DetailsWrapper>
+      <StyledNameId>{name ? `#${id}: ${name}` : `#${id}`}</StyledNameId>
+      <LabelsWrapper>
+        <StyledOptionLabel>{details.label}</StyledOptionLabel>
+        {details.filename && (
+          <StyledFileNameWrapper>
+            {'('}
+            <StyledFileName>{details.filename}</StyledFileName>
+            {')'}
+          </StyledFileNameWrapper>
+        )}
+      </LabelsWrapper>
+    </DetailsWrapper>
+    {crashed && (
+      <StyledCrashIcon
+        src="icon-warning-sm"
+        title={crashedInfo ? `(crashed with ${crashedInfo.values[0].type})` : ''}
+      />
+    )}
+  </Wrapper>
+);
+
+export default ThreadsSelectorOption;
+
+const Wrapper = styled('div')`
+  grid-template-columns: 1fr 30px;
+  display: grid;
+  align-items: center;
+`;
+
+const DetailsWrapper = styled('div')`
+  max-width: 100%;
+  overflow: hidden;
+  display: grid;
+  @media (min-width: ${props => props.theme.breakpoints[0]}) {
+    grid-template-columns: 100px 1fr;
+  }
+`;
+
+const LabelsWrapper = styled('div')`
+  display: grid;
+  width: 100%;
+  grid-gap: ${space(0.5)};
+  @media (min-width: ${props => props.theme.breakpoints[0]}) {
+    grid-template-columns: 1fr 200px;
+  }
+`;
+
+const StyledNameId = styled(TextOverflow)`
+  max-width: 100%;
+  text-align: left;
+  @media (min-width: ${props => props.theme.breakpoints[0]}) {
+    padding-right: ${space(1)};
+  }
+`;
+
+const StyledCrashIcon = styled(InlineSvg)`
+  color: #ec5e44;
+  margin-left: ${space(1)};
+`;
+
+const StyledFileNameWrapper = styled(TextOverflow)`
+  color: ${props => props.theme.purple};
+  display: flex;
+  text-align: left;
+  overflow: hidden;
+`;
+
+const StyledFileName = styled(StyledNameId)`
+  color: ${props => props.theme.purple};
+  padding-right: ${space(0)};
+`;
+
+// TODO(style): color not yet in the theme
+const StyledOptionLabel = styled(StyledNameId)`
+  color: #2c58a8;
```

Comment:
Could we use a theme color here? @doralchan is planning on refactoring our color names and normalizing the color values.